pkg/native/serializers/serializer_cdx.go

```diff
@@ -17,22 +19,27 @@ import (
 
 var _ native.Serializer = &CDX{}
 
+// Precompiled regex for serialNumber validation
+var serialNumberPattern = regexp.MustCompile(`^urn:uuid:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`)
+
 const (
 	stateKey state = "cyclonedx_serializer_state"
 )
 
 type (
 	state string
 	CDX   struct {
-		version  string
-		encoding string
+		version              string
+		encoding             string
+		GenerateSerialNumber bool
 	}
 )
 
 func NewCDX(version, encoding string) *CDX {
 	return &CDX{
-		version:  version,
-		encoding: encoding,
+		version:              version,
+		encoding:             encoding,
+		GenerateSerialNumber: false,
```

Comment:
Mmh no, sorry... Now that I look at it, the cdx options are not initialized in the serializer. Sorry for the wrong comment in the previous review. 

Let me open a PR to set them up so that you can use them here.

Reply:
OK, here: https://github.com/protobom/protobom/pull/327

Feel free to cherry-pick that commit if it helps unblocking you (and I'll close 327).

Reply:
Awesome, we merged #327. Just rebase and use the new option to control the serial number generation.

Reply:
done !!